warehouse/templates/accounts/register.html

```diff
@@ -70,46 +70,50 @@ <h1 class="page-title">Create an account on PyPI</h1>
-          </div>
-          <p class="form-group__help-text">
-            Choose a strong password that contains letters (uppercase and lowercase), numbers and special characters. Avoid common words or repetition.
-          </p>
-          <p class="form-group__help-text">
-            <strong>Password strength:</strong>
-            <span class="password-strength">
+        <div data-controller="password">
+          <div class="form-group">
+            <div class="split-layout">
+              <label for="password" class="form-group__label">Password</label>
+              <label for="show-password">
+                <input data-action="click->password#showPasswords" data-target="password.showPassword"
+                       id="show-password" type="checkbox">&nbsp;Show passwords
+              </label>
+            </div>
+            <div>
+              {{ form.new_password(placeholder="Select a password", required="required", class_="form-group__input", autocomplete="new-password", **{ 'data-target': 'password.newPassword' }) }}
+            </div>
+            <p class="form-group__help-text">
+              Choose a strong password that contains letters (uppercase and lowercase), numbers and special characters.
+              Avoid common words or repetition.
+            </p>
+            <p class="form-group__help-text">
+              <strong>Password strength:</strong>
+              <span class="password-strength">
               <span class="password-strength__gauge">
                 <span class="sr-only">Password field is empty</span>
               </span>
             </span>
-          </p>
-          {% if form.new_password.errors %}
-          <ul class="form-errors">
-            {% for error in form.new_password.errors %}
-            <li>{{ error }}</li>
-            {% endfor %}
-          </ul>
-          {% endif %}
-        </div>
+            </p>
+            {% if form.new_password.errors %}
+            <ul class="form-errors">
+              {% for error in form.new_password.errors %}
+              <li>{{ error }}</li>
+              {% endfor %}
+            </ul>
+            {% endif %}
+          </div>
 
-        <div class="form-group">
-          <label for="password_confirm" class="form-group__label">Confirm Password</label>
-          {{ form.password_confirm(placeholder="Confirm password", required="required", class_="form-group__input") }}
-          {% if form.password_confirm.errors %}
-          <ul class="form-errors">
-            {% for error in form.password_confirm.errors %}
-            <li>{{ error }}</li>
-            {% endfor %}
-          </ul>
-          {% endif %}
+          <div class="form-group">
+            <label for="password_confirm" class="form-group__label">Confirm Password</label>
+            {{ form.password_confirm(placeholder="Confirm password", required="required", class_="form-group__input", **{ 'data-target': 'password.passwordConfirm' }) }}
```

Comment:
WTForms will automatically change underscores to dashes in parameters it doesn't recognize, so this could just be `data_target="password.passwordConfirm"`